server/src/main/java/io/crate/auth/Credentials.java

```diff
@@ -23,9 +23,68 @@
 
 import java.io.Closeable;
 import org.elasticsearch.common.settings.SecureString;
+import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
-public record Credentials(@Nullable String username, @Nullable SecureString password) implements Closeable {
+/**
+ * Holder for all authentication methods.
+ * username is CrateDB user. Set up directly on password, cert and trust methods and resolved later on jwt.
+ * password is used only for password method and null otherwise.
+ * jwtToken is used only for jwt method and null otherwise.
+ */
+public class Credentials implements Closeable {
+
+    // Non-final as we set it up one we resolve CrateDB user.
+    private String username;
+
+    // Non-final as Postgres protocol might inject password later after creation.
+    private SecureString password;
+
+    private final String jwtToken;
+
+    private Credentials(@Nullable String username, @Nullable char[] password, @Nullable String jwtToken) {
+        this.username = username;
+        this.password = password != null ? new SecureString(password) : null;
+        this.jwtToken = jwtToken;
+    }
+
+    public static Credentials of(@NotNull String username, @Nullable char[] password) {
```

Comment:
minor: Could be a normal constructor as well, I see no reason for a factory method...

Reply:
updated to regular ctor